### PR TITLE
Add irrigation to rmf calculations

### PR DESCRIPTION
**Pull request by FrankvRaffe**
In addition to precipitation and evapotranspiration, add potential irrigation input to determine soil moisture deficit

## Summary by CodeRabbit

* **New Features**
  * Optional irrigation input now influences monthly water balance and simulation results.

* **Changes**
  * Weather input now includes a monthly reference ET plus an ET‑conversion factor (defaults to 0.75); soil moisture deficit incorporates irrigation.
  * Simulation outputs: psoc returned alongside soc; cstock option standardized to lowercase.
  * Default initialization now enabled.
  * Public parameter dataset renamed to rc_params.

* **Documentation**
  * Docs updated for irrigation and new weather fields.

* **Tests**
  * Expanded coverage for irrigation, ET factor, validation, and edge cases.

### CODE REVIEW

File: R/rc_input_rmf.R [hunk 1]
```diff
@@ -77,12 +107,11 @@ rc_input_rmf <- function(dt = NULL, B_DEPTH = 0.3, A_CLAY_MI,  dt.weather, dt.ti
   dt[, tsmdmax_cor := fifelse(crop_cover==1,tsmdmax,tsmdmax/1.8)]
 
   # Calculate actual evapotranspiration for months where only potential is provided (general rothc calculation)
-  dt[is.na(W_ET_ACT_MONTH), W_ET_ACT_MONTH := W_ET_POT_MONTH * 0.75]
+  dt[is.na(W_ET_ACT_MONTH), W_ET_ACT_MONTH := W_ET_POT_MONTH * W_POT_TO_ACT]
```
[the remark this answers is not in the document]
Comment (FrankvRaffe) [reply R/rc_input_rmf.R]: @coderabbitai adapted in latest commit

File: R/rc_input_rmf.R [hunk 2]
```diff
@@ -49,23 +66,36 @@ rc_input_rmf <- function(dt = NULL, B_DEPTH = 0.3, A_CLAY_MI,  dt.weather, dt.ti
...
+   # Derive year and month from irrigation data
+   dt.irrigation[, year := year(B_DATE_IRRIGATION)]
+   dt.irrigation[,month := month(B_DATE_IRRIGATION)]
+  
+   # Merge irrigation and crop cover data
+   dt <- merge(dt, dt.irrigation, by = c('year', 'month'), all.x = TRUE)[, B_IRR_AMOUNT := fifelse(is.na(B_IRR_AMOUNT), 0, B_IRR_AMOUNT)]
+      }else{
+        # Set irrigation inputs to 0
+      dt[, B_IRR_AMOUNT:= 0]  
+      }
```
[the remark this answers is not in the document]
Comment (FrankvRaffe) [reply R/rc_input_rmf.R]: @coderabbitai adapted in latest commit

File: R/rc_input_rmf.R [hunk 3]
```diff
@@ -22,24 +23,40 @@
...
+  if(!is.null(dt)){
   checkmate::assert_true(all(c('B_LU_START', 'B_LU_END') %in% colnames(dt)))
   checkmate::assert_date(as.Date(dt$B_LU_START), any.missing = F)
   checkmate::assert_date(as.Date(dt$B_LU_END), any.missing = F)
+  }
   checkmate::assert_data_table(dt.weather, null.ok = FALSE)
-  checkmate::assert_subset(c("month","W_TEMP_MEAN_MONTH","W_PREC_SUM_MONTH"), colnames(dt.weather))
+  checkmate::assert_subset(colnames(dt.weather), choices = c("year", "month", "W_TEMP_MEAN_MONTH", "W_PREC_SUM_MONTH", "W_ET_POT_MONTH", "W_ET_ACT_MONTH", "W_POT_TO_ACT"))
   checkmate::assert(any(c("W_ET_POT_MONTH","W_ET_ACT_MONTH") %in% colnames(dt.weather)),
                          msg = "At least one of 'W_ET_POT_MONTH' or 'W_ET_ACT_MONTH' must be provided.")
```
[the remark this answers is not in the document]
Comment (FrankvRaffe) [reply R/rc_input_rmf.R]: @coderabbitai excellent suggestion, adapted in latest commit. Additionally added W_TEMP_MEAN_MONT and W_PREC_SUM_MONTH as additional requirements

File: R/rc_helpers.R [hunk 4]
```diff
@@ -33,7 +33,12 @@ cf_ind_importance <- function(x) {
 #' @export
 #'
 rc_update_weather <- function(dt = NULL){
+  # Add visible bindings
+  W_POT_TO_ACT = NULL
```
Comment (SvenVw) [inline R/rc_helpers.R]: This naming is confusing. I would suggest something like W_ET_FACTOR_POTACT or something similiar. @BrentWHR what would you suggest?

**Review by SvenVw — COMMENTED** [review]
LGTM, just check if we can improve the naming of  W_POT_TO_ACT before merging

File: tests/testthat/test-rc_helpers.R [hunk 5]
```diff
@@ -716,3 +717,168 @@ test_that("rc_time_period handles edge cases", {
...
+    month = 1:12,
+    W_TEMP_MEAN_MONTH = rep(10, 12),
+    W_PREC_SUM_MONTH = rep(50, 12),
+    W_ET_POT_MONTH = rep(50, 12)
+  )
+  
+  result3 <- rc_update_weather(weather_no_correction)
+  expect_true("W_POT_TO_ACT" %in% names(result3))
+  expect_equal(result3$W_POT_TO_ACT, rep(0.75, 12))
+})
```
[the remark this answers is not in the document]
Comment (FrankvRaffe) [reply tests/testthat/test-rc_helpers.R]: Agreed! I have split it into several tests with appropriate naming now.

File: tests/testthat/test-rc_helpers.R [hunk 6]
```diff
@@ -14,16 +14,16 @@ test_that("rc_update_weather validates input data table", {
...
   )
   
   # Test with valid data table
   expect_no_error(rc_update_weather(valid_dt))
   
   # Test missing columns
-  invalid_dt <- valid_dt[, W_ET_POT_MONTH := NULL]
-  expect_no_error(rc_update_weather(invalid_dt)) # only one of W_ET_POT_MONTH or W_ET_ACT_MONTH must be provided
+  invalid_dt <- valid_dt[, W_ET_REF_MONTH := NULL]
+  rc_update_weather(invalid_dt) # only one of W_ET_REF_MONTH or W_ET_ACT_MONTH must be provided
```
[the remark this answers is not in the document]
Comment (FrankvRaffe) [reply tests/testthat/test-rc_helpers.R]: @coderabbitai adapted!

File: R/rc_helpers.R [hunk 7]
```diff
@@ -33,7 +34,12 @@ cf_ind_importance <- function(x) {
...
+  
+  # Check if there is a weather data table
   if(!is.null(dt)){
```
[the remark this answers is not in the document]
Comment (BrentWHR) [reply R/rc_helpers.R]: Ah yes, good point, I overlooked this.

File: R/rc_input_rmf.R [hunk 8]
```diff
@@ -20,52 +21,103 @@
...
+   # Derive year and month from irrigation data, aggregate per month
+        irrig[, year := year(B_DATE_IRRIGATION)]
+        irrig[,month := month(B_DATE_IRRIGATION)]
+        dt.irr.month <- irrig[, .(B_IRR_AMOUNT = sum(B_IRR_AMOUNT, na.rm = TRUE)), by = .(year, month)]
+   
+  
+   # Merge irrigation and crop cover data
+   dt <- merge(dt, dt.irr.month, by = c('year', 'month'), all.x = TRUE)[
+     , B_IRR_AMOUNT := fifelse(is.na(B_IRR_AMOUNT), 0, B_IRR_AMOUNT)]
+      }else{
```
[the remark this answers is not in the document]
Comment (FrankvRaffe) [reply R/rc_input_rmf.R]: @coderabbit implemented in latest commit

**Review by BrentWHR — APPROVED** [review]
LGTM
